storage/src/mmr/journaled.rs

```diff
@@ -124,20 +191,46 @@ impl<B: Blob, E: RStorage<B> + Metrics, H: Hasher> Mmr<B, E, H> {
 
     /// Sync any new elements to disk.
     pub async fn sync(&mut self) -> Result<(), Error> {
```

Comment:
nit: it feels odd to me that in `sync` you update `metadata` -> `journal` but in `close` you close `journal` -> `metadata`.

Reply:
Note that close() calls sync up top, so technically they both do metadata->journal for flushing data to disk which is the necessary order for ensuring recovery.  close() does another metadata write afterwards for the purpose of clearing out historical peaks, but this is just to free up space and isn't actually required. (In fact I'm investigating if we should hold on to historical peaks for longer to make the min-provable position stable.)

Reply:
now that we sync journal first, was able to remove the peak-clearing logic from close()